phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java

```diff
@@ -230,7 +231,13 @@ public static void deleteTask(PhoenixConnection conn, PTable.TaskType taskType,
                 taskQuery += " AND " + PhoenixDatabaseMetaData.TASK_DATA + " LIKE '%" + indexName + "%'";
             }
 
-        return populateTasks(connection, taskQuery);
+            List<TaskRecord> result = populateTasks(connection, taskQuery);
+            if (ts != null) {
+                // Adding TASK_TS to the where clause did not work. It returns empty when directly querying with the timestamp.
+                result.removeIf(p -> !p.getTimeStamp().equals(ts));
```

Comment:
This might complain for some branches which need to be source compatible with Java 7 instead of 8